docker-compose.yml

```diff
@@ -1,136 +1,155 @@
+        networks:
+            - omero-network
+        volumes:
+            - ./server:/home/omero
+        environment:
+            - SLAVE_NAME=omero
+            - SLAVE_EXECUTORS=2
+            - SLAVE_PARAMS=-labels centos7 -labels ice36 -labels java18 -disableClientsUniqueId
+            - JENKINS_MASTER=http://jenkins:8080
+        ports:
+            - "4064"
+            - "4063"
 
-nginxjenkins:
-    image: nginx:1.10
-    links:
-        - jenkins
-    volumes:
-        - ./jenkins/conf.d:/etc/nginx/conf.d
-        - ./jenkins/sslcert:/etc/nginx/ssl
-    ports:
-        - "443"
-redis:
-    image: redis
+    web:
+        build:
+            context: ./web
+            dockerfile: Dockerfile
+        networks:
+            - omero-network
+        volumes:
+            - ./web:/home/omero
+            - ./nginx/conf.d:/home/omero/nginx
+        environment:
+            - SLAVE_NAME=web
+            - SLAVE_EXECUTORS=2
+            - SLAVE_PARAMS= -labels centos7 -labels ice36 -labels java18 -disableClientsUniqueId
+            - JENKINS_MASTER=http://jenkins:8080
+            - WEB_PREFIX=/web
 
-seleniumhub:
-    image: selenium/hub:3.0.0-dubnium
-    environment:
-        - GRID_BROWSER_TIMEOUT=60
-        - GRID_TIMEOUT=60
-        - GRID_MAX_SESSION=2
-        #- SE_OPTS=-browserTimeout 60 -timeout 60 maxSession 2
-#    volumes:
-#        - /dev/urandom:/dev/random
-    ports:
-        - "4444"
-seleniumfirefox:
-    image: selenium/node-firefox:3.0.0-dubnium
-    links:
-        - seleniumhub:hub
-#    volumes:
-#        - /dev/shm:/dev/shm
-seleniumchrome:
-    image: selenium/node-chrome:3.0.0-dubnium
-    links:
-        - seleniumhub:hub
-#    volumes:
-#        - /dev/shm:/dev/shm
+    nginx:
+        build:
+            context: ./nginx
+            dockerfile: Dockerfile
+        networks:
+            - omero-network
+        volumes:
+            - ./nginx/conf.d:/etc/nginx/conf.d
+            - ./web/static:/home/omero/static
+            - ./nginx/log:/var/log/nginx
+            - ./nginx/sslcert:/etc/nginx/ssl
+        environment:
+            - SLAVE_NAME=nginx
+            - SLAVE_PARAMS=-labels centos7 -labels java18 -disableClientsUniqueId
+            - JENKINS_MASTER=http://jenkins:8080
+        ports:
+            - "80"
+            - "443"
+
+    nginxjenkins:
+        image: nginx:1.10
+        volumes:
+            - ./jenkins/conf.d:/etc/nginx/conf.d
+            - ./jenkins/sslcert:/etc/nginx/ssl
+        ports:
+            - "443"
+        networks:
+            - omero-network
+
+    redis:
+        image: redis
+        networks:
+            - omero-network
+
+    seleniumhub:
+        image: selenium/hub
+        environment:
+            - GRID_TIMEOUT=0
+        ports:
+            - "4444"
+        networks:
+            - omero-network
+
+    seleniumfirefox:
+        image: selenium/node-firefox
+        environment:
+            - HUB_PORT_4444_TCP_ADDR=devspace_seleniumhub_1
```

Comment:
Is this env-var needed? If so setting it to `seleniumhub` should work, the name `devspace_seleniumhub_1` may vary with different deployments.